### ISSUE TITLE
Jasmine tests dont seem to be working

### ISSUE THREAD

**Issue by frankmt** [fdietz/team_dashboard#17]
Hi,

Im trying to run the jasmine specs but they dont seem to be passing as from today. Since I'm not completely sure if they are being maintained or not (or if there is a build somewhere), just wanted to check they should be ok.

Also would be interested in contributing to the project, but havent found any roadmap or list of issues so I can see if the things I want to do are planned or not.

**Comment by fdietz** [comment]
Hi Francisco,

thanks for looking into Team Dashboard!

I've actually just fixed the jasmine javascript tests today :-)

I've used the github issue tracker to create items I'd like to focus on
(for milestone v0.3 and v0.4). Since the v0.4 will be released this week,
I've already completed all of them. I will create tickets for the next
release this week for you to look into.

When looking into where to contribute, we should discuss things in the
mailinglist (https://groups.google.com/forum/#!forum/team_dashboard), newly
created for just that purpose.

I'm completely open here. So, just send me your ideas and what you would
like to work on and we discuss what makes sense!

Looking forward to hearing from you again!

Cheers,
-Frederik

**Comment by fdietz** [comment]
Hi again,

you might wanna check out the issues I've created for milestone v0.5 here:
https://github.com/fdietz/team_dashboard/issues?milestone=3&page=1&state=open

Please, let me know if you interested in something here!

Cheers,
-Frederik

**Comment by frankmt** [comment]
Thanks!

Will do, going to try to put some time into it over the weekend.

Cheers
Francisco

**Comment by frankmt** [comment]
Hi Frederik,

Ive taken a look at the issues and could work on this one till the end of
the week if no one else is at the moment:

https://github.com/fdietz/team_dashboard/issues/21

Ive also created a new issue for something that would be useful for my team
at the moment (https://github.com/fdietz/team_dashboard/issues/30), please
let me know what you think.

Cheers
Francisco

On Fri, Sep 7, 2012 at 8:38 AM, Francisco Trindade <frank.trindade@gmail.com

> wrote:
> 
> Thanks!
> 
> Will do, going to try to put some time into it over the weekend.
> 
> Cheers
> Francisco
> 
> Sent from my iPhone
> 
> On 06/09/2012, at 11:43 PM, Frederik Dietz notifications@github.com
> wrote:
> 
> Hi again,
> 
> you might wanna check out the issues I've created for milestone v0.5 here:
> 
> https://github.com/fdietz/team_dashboard/issues?milestone=3&page=1&state=open
> 
> Please, let me know if you interested in something here!
> 
> Cheers,
> -Frederik
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/fdietz/team_dashboard/issues/17#issuecomment-8332464.

## 

## 

Francisco Trindade

**Comment by fdietz** [comment]
Hi Francisco,

awesome! Nobody's working on item #21 yet. So go ahead and let me know if
you want to discuss this in more detail!

Really like your proposal (#30)! Would be great to have overlay periods
like last year, last month, yesterday over todays data.

Cheers,
-Frederik

**Comment by frankmt** [comment]
Hi Frederick,

I haven't had time to look at Teheran issue this week, but just to let you know I havent given up.

I'm reserving Monday night to work on the dashboard, so you will hear more news then.

Nice that you liked #30, if it's still not done I will work on that after #21.

Cheers
Francisco

**Comment by fdietz** [comment]
Hi Francisco,

no pressure ;-)

I've followed up on a lots of things concerning the current graph rendering
and came up with some ideas I want to share with you. I assume it might
affect how you realize your solution.

I've found several problems with rickshaw.js:
- it can't render two metrics in the same graph with a different total
  count of datapoints
- it can't render metrics which contain null values (I had to actually
  replace these with Integer 0 values which is unusable for some of the
  scenarios in my company)
- people want an improved y-axis/y-axis labeling

I've played around with a lot of graphing js libraries and found flotr2 (
http://humblesoftware.com/flotr2/) works brilliantly and solves both
problems. I've created therefore a feature branch flotr2 and will spend
some time this weekend integration it. I will leave the rickshaw.js
implementation intact (so we can easily switch back and forth), because I
really like it and probably will move back using it when my problems are
addressed.

Things we can't do in the graph right now, is graphite functions are for
example constantLine. This will draw a horizontal line in the graph, but
graphites json api only returns a single datapoint. I suggest to open an
issue for the graphite-web project to change the json response.

That leaves us with events (and graphites drawAsInfinite function), which
are displayed in the graph as a vertical line (maybe with annotation). Both
graphite and ganglia json apis are incomplete again here. They don't return
events datapoints at all. So, I guess we should consider treating events in
team dashboard as first class citizens allowing events to be overlays to
graphs.

So, to sum everything up it would be awesome if you wrap your head around
how to handle events. Here are some pointers how others are doing it:

http://blog.vuksan.com/2010/06/28/overlay-deploy-timeline-on-your-ganglia-graphs/
http://ganglia.info/?p=382
http://graphite.readthedocs.org/en/0.9.9/functions.html (graphite
drawAsInfinite function reference)
https://github.com/stumbleupon/opentsdb/issues/16

Let me know what you think!

Cheers,
-Frederik

**Comment by frankmt** [comment]
Sounds great. Ive spent some time looking at the code today and will do
more tmrw and monday.

I assume its best if I start working in the flotr branch. Is that correct?

Another thing I wanted to ask you, is how you are testing things locally.
Do you run ganglia and graphite on your box?

Cheers
Francisco

On Fri, Sep 14, 2012 at 5:10 PM, Frederik Dietz notifications@github.comwrote:

> Hi Francisco,
> 
> no pressure ;-)
> 
> I've followed up on a lots of things concerning the current graph
> rendering
> and came up with some ideas I want to share with you. I assume it might
> affect how you realize your solution.
> 
> I've found several problems with rickshaw.js:
> - it can't render two metrics in the same graph with a different total
>   count of datapoints
> - it can't render metrics which contain null values (I had to actually
>   replace these with Integer 0 values which is unusable for some of the
>   scenarios in my company)
> - people want an improved y-axis/y-axis labeling
> 
> I've played around with a lot of graphing js libraries and found flotr2 (
> http://humblesoftware.com/flotr2/) works brilliantly and solves both
> problems. I've created therefore a feature branch flotr2 and will spend
> some time this weekend integration it. I will leave the rickshaw.js
> implementation intact (so we can easily switch back and forth), because I
> really like it and probably will move back using it when my problems are
> addressed.
> 
> Things we can't do in the graph right now, is graphite functions are for
> example constantLine. This will draw a horizontal line in the graph, but
> graphites json api only returns a single datapoint. I suggest to open an
> issue for the graphite-web project to change the json response.
> 
> That leaves us with events (and graphites drawAsInfinite function), which
> are displayed in the graph as a vertical line (maybe with annotation).
> Both
> graphite and ganglia json apis are incomplete again here. They don't
> return
> events datapoints at all. So, I guess we should consider treating events
> in
> team dashboard as first class citizens allowing events to be overlays to
> graphs.
> 
> So, to sum everything up it would be awesome if you wrap your head around
> how to handle events. Here are some pointers how others are doing it:
> 
> http://blog.vuksan.com/2010/06/28/overlay-deploy-timeline-on-your-ganglia-graphs/
> http://ganglia.info/?p=382
> http://graphite.readthedocs.org/en/0.9.9/functions.html (graphite
> drawAsInfinite function reference)
> https://github.com/stumbleupon/opentsdb/issues/16
> 
> Let me know what you think!
> 
> Cheers,
> -Frederik
> 
> On Fri, Sep 14, 2012 at 12:01 AM, Francisco Trindade <
> notifications@github.com> wrote:
> 
> > Hi Frederick,
> > 
> > I haven't had time to look at Teheran issue this week, but just to let
> > you
> > know I havent given up.
> > 
> > I'm reserving Monday night to work on the dashboard, so you will hear
> > more
> > news then.
> > 
> > Nice that you liked #30, if it's still not done I will work on that
> > after
> > #21.
> > 
> > Cheers
> > Francisco
> > 
> > Sent from my iPhone
> > 
> > On 11/09/2012, at 1:30 AM, Frederik Dietz notifications@github.com
> > wrote:
> > 
> > > Hi Francisco,
> > > 
> > > awesome! Nobody's working on item #21 yet. So go ahead and let me know
> > > if
> > > you want to discuss this in more detail!
> > > 
> > > Really like your proposal (#30)! Would be great to have overlay
> > > periods
> > > like last year, last month, yesterday over todays data.
> > > 
> > > Cheers,
> > > -Frederik
> > > 
> > > On Mon, Sep 10, 2012 at 2:50 PM, Francisco Trindade <
> > > notifications@github.com> wrote:
> > > 
> > > > Hi Frederik,
> > > > 
> > > > Ive taken a look at the issues and could work on this one till the
> > > > end
> > > > of
> > > > the week if no one else is at the moment:
> > > > 
> > > > https://github.com/fdietz/team_dashboard/issues/21
> > > > 
> > > > Ive also created a new issue for something that would be useful for
> > > > my
> > > > team
> > > > at the moment (https://github.com/fdietz/team_dashboard/issues/30),
> > > > please
> > > > let me know what you think.
> > > > 
> > > > Cheers
> > > > Francisco
> > > > 
> > > > On Fri, Sep 7, 2012 at 8:38 AM, Francisco Trindade <
> > > > frank.trindade@gmail.com
> > > > 
> > > > > wrote:
> > > > > 
> > > > > Thanks!
> > > > > 
> > > > > Will do, going to try to put some time into it over the weekend.
> > > > > 
> > > > > Cheers
> > > > > Francisco
> > > > > 
> > > > > Sent from my iPhone
> > > > > 
> > > > > On 06/09/2012, at 11:43 PM, Frederik Dietz <
> > > > > notifications@github.com>
> > > > > 
> > > > > wrote:
> > > > > 
> > > > > Hi again,
> > > > > 
> > > > > you might wanna check out the issues I've created for milestone
> > > > > v0.5
> > > > > here:
> > 
> > https://github.com/fdietz/team_dashboard/issues?milestone=3&page=1&state=open
> > 
> > > > > Please, let me know if you interested in something here!
> > > > > 
> > > > > Cheers,
> > > > > -Frederik
> > > > > 
> > > > > —
> > > > > Reply to this email directly or view it on GitHub<
> > > > 
> > > > https://github.com/fdietz/team_dashboard/issues/17#issuecomment-8332464>.
> > > > 
> > > > 
> > > > ## 
> > > > 
> > > > ## 
> > > > 
> > > > Francisco Trindade
> > > > 
> > > > —
> > > > Reply to this email directly or view it on GitHub<
> > > > https://github.com/fdietz/team_dashboard/issues/17#issuecomment-8422334>.
> > > > 
> > > > —
> > > > Reply to this email directly or view it on GitHub.
> > 
> > —
> > Reply to this email directly or view it on GitHub<
> > https://github.com/fdietz/team_dashboard/issues/17#issuecomment-8545182>.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/fdietz/team_dashboard/issues/17#issuecomment-8553297.

## 

## 

Francisco Trindade

**Comment by fdietz** [comment]
Hi again!

I've actually already merged the flotr branch into master. Just deleted it
know in order to not confuse anyone.

I've installed graphite on a virtualbox with Ubuntu Linux - never managed
to install it on OS X. Here's the gist:
https://gist.github.com/2820069

Also, my company is using graphite and ganglia setup by site ops. So, its
easy for me to test things in a much larger scale. For your tests its
probably sufficient to start with graphite. In my company we use graphite
as the "graphing engine" behind ganglia anyways.

I'd love to see the datapoints source plugins to support a "marker" for
events such as "deployment" or "cronjob".
So, adding a marker attribute would probably make sense and gives us a nice
abstraction to cover inconsistencies between various systems.

example:
[
{ :target => "demo.example1", :datapoints => \* [[1, 1234], [4, 1244]],
:markers => [1240, 1300]\* },
]

So, the markers attribute is an array with timestamps. Hope you get the
idea!

BTW, I will be on vacation starting tomorrow for two weeks, probably with
no good internet access.

Cheers,
-Frederik
